src/components/TopContainer/TopContainer.jsx

```diff
@@ -0,0 +1,52 @@
+import React, { useEffect, useRef } from "react";
+import BgImage from "../../assets/hero-img.jpg";
+import MovieDetails from "../MovieDetails/MovieDetails";
+import {connect} from "react-redux";
+import "./top-container.scss"
+import SearchForm from "../SearchForm/SearchForm";
+import { useParams } from "react-router-dom";
+import { getMovie} from "../../store/movie/action-creators";
+
+function usePrevious(value) {
+    const ref = useRef();
+    useEffect(() => {
+        ref.current = value;
+    });
+    return ref.current;
+}
+
+const TopContainer = (props) => {
+    const {id} = useParams();
+    const prevId = usePrevious(id);
+
+    useEffect(() => {
+        if (id && id !== prevId) {
+            props.getMovieDetails(id);
+        }
+    }, [id, prevId]);
```

Comment:
I might be wrong but shouldn't it work without `prevId`? if there is another id in url I believe it should trigger current `useEffect`, shouldn't it? 🤔

Reply:
yep)